lighthouse-assert/assert/assert.ts

```diff
@@ -0,0 +1,405 @@
+export interface IResult {
```

Comment:
this feels a bit like a monolith file to have so many types, wdyt about breaking out the results to an object that would be returned by an `Assert` instance and moving some of the logic there?

Reply:
Totally agree, I'll split it up.

> breaking out the results to an object that would be returned by an Assert instance and moving some of the logic there

do you mean all 'collate' logic?

Reply:
yeah basically the same idea as https://github.com/GoogleChrome/lighthouse/pull/3283#discussion_r137894453

if `collate()` returned some `ExpectationsResult` object that had the `isEqual()` and `getDifferenceCount()` methods on it